pkg/terminal/command.go

```diff
@@ -2633,6 +2638,48 @@ func (c *Commands) sourceCommand(t *Term, ctx callContext, args string) error {
 	return c.executeFile(t, args)
 }
 
+func (c *Commands) saveStateCommand(t *Term, ctx callContext, args string) error {
+	if len(args) == 0 {
+		return errors.New("wrong number of arguments: source <filename>")
+	}
+	breakPoints, err := t.client.ListBreakpoints(args == "-a")
+	if err != nil {
+		return err
+	}
+	var fileContent string
```

Comment:
Will be more efficient to use a `strings.Builder` here: https://pkg.go.dev/strings#Builder (I know this is a WIP just doing an initial review pass anyways).

Reply:
Alternatively `w := bufio.NewWriter(file)` and then write everything to `w`.

Reply:
I followed @aarzilli suggestion and used `bufio.NewWriter`.